src/OpenTelemetry.Api/Trace/Source.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="Source.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Controls the source name and version that will be analyzed.
+    /// </summary>
+    public struct Source
+    {
+        /// <summary>
+        /// Source name.
+        /// </summary>
+        public readonly string Name;
+
+        /// <summary>
+        /// Min version of the assembly.
+        /// </summary>
+        public readonly Version MinVersion;
+
+        /// <summary>
+        /// Max version of the assembly.
+        /// </summary>
+        public readonly Version MaxVersion;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Source"/> struct.
+        /// </summary>
+        /// <param name="name">Name of the source.</param>
+        /// <param name="minVersion">Min version.</param>
+        /// <param name="maxVersion">Max version.</param>
+        public Source(string name, Version minVersion = null, Version maxVersion = null)
```

Comment:
should we force users to provider Version or just  a string? internally we can store it as version. It'll lot easy for users to provider "1.0.0" string

Reply:
i like and dislike...yes, it would be able to facilitate, but users could break when adding to source. So, to use that, I think I prefer to break the user first instead of in our method....What do you think?

Reply:
if the user pass: `abc123`, it would break. If he creates the version before, he would see that that version is invalid. but, both cases should work :)

Reply:
We can throw from this method, if the string they pass is not parseable into Version.

Reply:
ok. let me do that change